MaxText/layers/linears.py

```diff
@@ -172,12 +211,19 @@ def __call__(self, inputs, decode: bool = False, deterministic: bool = False):
             kernel_axes=('embed', 'mlp'),
             name=dense_name,
             use_int8=cfg.int8_training,
+            use_bias=self.use_bias,
         )(inputs)
         x = _convert_to_activation_function(act_fn)(x)
         activations.append(x)
 
     # Take elementwise product of above intermediate activations.
     x = functools.reduce(operator.mul, activations)
+
+    if self.apply_padding_mask and padding_mask is not None:
+      # from [B, L] to [B, L, D]
+      padding_mask = jnp.expand_dims(padding_mask, axis=-1)
```

Comment:
Sorry what is the padding mask encoding? Not sure I follow

Reply:
I think it is used to mask out padding in calculating representations of each token.
[Original source](https://github.com/google/praxis/blob/fbde0b77ff4a77fb12f05ffaf2adce9a4c7806ed/praxis/layers/transformers.py#L467)

For example:
```
inputs = [[11,12,0,0]]
padding_mask = [[1, 1, 0, 0]]
output *= padding_mask

# The 2nd half of activations/output will all be zeros.
```

Reply:
This explanation isn't very satisfying. Don't those entries have zero impact in attention? And the MLPs are separable. How are they interacting?

Reply:
Discussed offline, will remove paddings.